packages/react-aria-components/stories/Tree.stories.tsx

```diff
@@ -295,25 +316,161 @@ WithLinks.story = {
   }
 };
 
-interface ItemType {
-  childItems: Iterable<object>,
-  name: string
+function renderEmptyLoader({isLoading}) {
+  return isLoading ? 'Root level loading spinner' : 'Nothing in tree';
 }
-export const EmptyTree = (args: TreeProps<unknown>) => (
+
+const EmptyTreeStatic = (args: {isLoading: boolean}) => (
   <UNSTABLE_Tree
     {...args}
     className={styles.tree}
-    aria-label="test dynamic tree"
-    items={[]}
-    renderEmptyState={() => <span>Nothing in tree</span>}>
-    {(item: ItemType) => (
-      <DynamicTreeItem childItems={item?.childItems} textValue={item?.name}>
-        {item?.name}
-      </DynamicTreeItem>
-    )}
+    aria-label="test empty static tree"
+    renderEmptyState={() => renderEmptyLoader({isLoading: args.isLoading})}>
+    <Collection items={[]} dependencies={[args.isLoading]}>
+      {(item: any) => (
+        <DynamicTreeItem renderLoader={(id) => id === 'project-2C'} isLoading={args.isLoading} id={item.id} childItems={item.childItems} textValue={item.name}>
+          {item.name}
+        </DynamicTreeItem>
+      )}
+    </Collection>
   </UNSTABLE_Tree>
 );
 
+export const EmptyTreeStaticStory = {
+  render: EmptyTreeStatic,
+  args: {
+    isLoading: false
+  },
+  name: 'Empty/Loading Tree rendered with TreeLoader collection element'
+};
+
+function LoadingStoryDepOnCollection(args) {
+  return (
+    <UNSTABLE_Tree {...args} defaultExpandedKeys={defaultExpandedKeys} disabledKeys={['reports-1AB']} className={styles.tree} aria-label="test dynamic tree" onExpandedChange={action('onExpandedChange')} onSelectionChange={action('onSelectionChange')}>
+      <Collection items={rows} dependencies={[args.isLoading]}>
+        {(item) => (
+          <DynamicTreeItem renderLoader={(id) => id === 'project-2C'} isLoading={args.isLoading} id={item.id} childItems={item.childItems} textValue={item.name}>
+            {item.name}
+          </DynamicTreeItem>
+        )}
+      </Collection>
+      {args.isLoading && <MyTreeLoader />}
+    </UNSTABLE_Tree>
+  );
+}
+
+export const LoadingStoryDepOnCollectionStory = {
+  render: LoadingStoryDepOnCollection,
+  args: {
+    isLoading: false
+  },
+  name: 'Loading, static root loader and dynamic rows'
```

Comment:
maybe add a description to this story and the ones that follow it? I'm not sure what I'm supposed to see or do with it